src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -187,9 +163,24 @@ private static PresenceData extractPresenceDataFrom(final String message) {
     }
 
     @SuppressWarnings("rawtypes")
-    private void storeMyUserId(final Object channelData) {
-        final Map channelDataMap = GSON.fromJson((String)channelData, Map.class);
-        myUserID = String.valueOf(channelDataMap.get("user_id"));
+    private String extractUserIdFromChannelData(final String channelData) {
+        final Map channelDataMap;
+        try {
+            channelDataMap = GSON.fromJson((String)channelData, Map.class);
+        } catch (final JsonSyntaxException e) {
+            throw new AuthorizationFailureException("Invalid response from Authorizer: unable to parse channel_data object: " + channelData, e);
+        }
+        Object maybeUserId;
+        try {
+            maybeUserId = channelDataMap.get("user_id");
```

Comment:
channelDataMap.containsValue("user_id")? ;)

Reply:
Thanks @kn100! That is a reasonable suggestion, but I don't think it's worth doing now.